### PR TITLE
Remove old cuDNN support

### PR DESCRIPTION
**Pull request by okuta**
This PR removes old cuDNN support (v2 and v3). This PR remove only unused code.

### CODE REVIEW

File: chainer/functions/pooling/average_pooling_nd.py [hunk 1]
```diff
@@ -46,16 +40,10 @@ def forward_cpu(self, x):
         return y,
 
     def forward_gpu(self, x):
-        if (chainer.should_use_cudnn('>=auto') and
-                pooling_nd._check_cudnn_acceptable_type(x[0].dtype)):
+        if chainer.should_use_cudnn('>=auto') and self.ndim >= 2:
```
Comment (unnonouno) [inline chainer/functions/pooling/average_pooling_nd.py]: Do we need to check types?

File: chainer/functions/pooling/max_pooling_2d.py [hunk 2]
```diff
@@ -32,8 +28,7 @@ def forward_cpu(self, x):
         return y,
 
     def forward_gpu(self, x):
-        if (chainer.should_use_cudnn('>=auto') and
-                pooling_2d._check_cudnn_acceptable_type(x[0].dtype)):
+        if chainer.should_use_cudnn('>=auto'):
```
Comment (unnonouno) [inline chainer/functions/pooling/max_pooling_2d.py]: ditto

File: chainer/functions/pooling/max_pooling_nd.py [hunk 3]
```diff
@@ -50,16 +44,10 @@ def forward_cpu(self, x):
         return y,
 
     def forward_gpu(self, x):
-        if (chainer.should_use_cudnn('>=auto') and
-                pooling_nd._check_cudnn_acceptable_type(x[0].dtype)):
+        if chainer.should_use_cudnn('>=auto') and self.ndim >= 2:
```
Comment (unnonouno) [inline chainer/functions/pooling/max_pooling_nd.py]: ditto

File: chainer/functions/pooling/average_pooling_nd.py (hunk 1 above)
Comment (okuta) [inline chainer/functions/pooling/average_pooling_nd.py]: No, cuDNN v3 and later accept all dtype.
```
def _check_cudnn_acceptable_type(x_dtype):
    return _cudnn_version >= 3000 or x_dtype != numpy.float16
```

Comment (unnonouno) [inline chainer/functions/pooling/average_pooling_nd.py]: ah, ok

**Comment by unnonouno** [comment]
LGTM

**Pull request merged**